tests/Integration/Behaviour/Features/Context/CarrierFeatureContext.php

```diff
@@ -0,0 +1,375 @@
+use Context;
+use Country;
+use Group;
+use RangePrice;
+use State;
+use Zone;
+
+class CarrierFeatureContext implements BehatContext
+{
+    use CartAwareTrait;
+
+    /**
+     * @var Zone[]
+     */
+    protected $zones = [];
+
+    /**
+     * @var Country[]
+     */
+    protected $countries = [];
+
+    /**
+     * @var Country[]
+     */
+    protected $previousCountries = [];
+
+    /**
+     * @var State[]
+     */
+    protected $states = [];
+
+    /**
+     * @var Address[]
+     */
+    protected $addresses = [];
+
+    /**
+     * @var Carrier[]
+     */
+    protected $carriers = [];
+
+    /**
+     * @var RangePrice[]
+     */
+    protected $priceRanges = [];
+
+    /**
+     * @var CustomerFeatureContext
+     */
+    protected $customerFeatureContext;
+
+    /** @BeforeScenario */
+    public function before(BeforeScenarioScope $scope)
+    {
+        $this->customerFeatureContext = $scope->getEnvironment()->getContext(CustomerFeatureContext::class);
+    }
+
+    /**
+     * @Given /^There is a zone with name (.+)$/
+     */
+    public function setZone($zoneName)
+    {
+        $zone = new Zone();
+        $zone->name = $zoneName;
+        $zone->add();
+        $this->zones[$zoneName] = $zone;
+    }
+
+    /**
+     * @param $zoneName
+     */
+    public function checkZoneWithNameExists($zoneName)
+    {
+        if (!isset($this->zones[$zoneName])) {
+            throw new \Exception('Zone with name "' . $zoneName . '" was not added in fixtures');
+        }
+    }
+
+    /**
+     * @Given /^There is a country with name (.+) and iso code (.+) in zone named (.+)$/
+     */
+    public function setCountry($countryName, $isoCode, $zoneName)
+    {
+        $this->checkZoneWithNameExists($zoneName);
+        $countryId = Country::getByIso($isoCode, false);
+        if (!$countryId) {
+            throw new \Exception('Country not found with iso code = ' . $isoCode);
+        }
+        $country = new Country($countryId);
+        // clone country to be able to properly reset previous data
+        $this->previousCountries[$countryName] = clone $country;
+        $this->countries[$countryName] = $country;
+        $country->id_zone = $this->zones[$zoneName]->id;
+        $country->active = 1;
+        $country->save();
+    }
+
+    /**
+     * @param $countryName
+     *
+     * @return Country
+     */
+    public function getCountryWithName($countryName)
+    {
+        return $this->countries[$countryName];
+    }
+
+    /**
+     * @param $countryName
+     */
+    public function checkCountryWithNameExists($countryName)
+    {
+        if (!isset($this->countries[$countryName])) {
+            throw new \Exception('Country with name "' . $countryName . '" was not added in fixtures');
+        }
+    }
+
+    /**
+     * @Given /^There is a state with name (.+) and iso code (.+) in country named (.+) and zone named (.+)$/
+     */
+    public function setState($stateName, $stateIsoCode, $countryName, $zoneName)
```

Comment:
Maybe `setState` is a bit vague, it does not indicate clearly whether you __create__ a State or you __edit__ a state